regression_tests/tests/gitbook/test_docs.py

```diff
@@ -0,0 +1,181 @@
+]
+
+blacklist_snippets = {}
+
+def get_code(page: str) -> List[str]:
+    '''
+    Use an regular expression to find all Python code blocks in the page.
+    Return
+    - List of each code block found in the page in order of appearance
+    '''
+    contents = Path(page).read_text()
+
+    # The regular expression is used to extract code blocks written in the Python language that are enclosed in a triple backtick (```) fence.
+
+    # - `{3} matches exactly three occurrences of the previous character or group, which in this case is the backtick (`).
+    # - python\n matches the string "python" followed by a newline character.
+    # - ([\s\S]*?) is a capture group that matches any sequence of characters, including whitespace characters and line breaks. The *? quantifier means "match zero or more of the preceding token, but as few as possible".
+    # - \n`{3} matches exactly a newline followed by three three backticks (`).
+    
+    # The regular expression matches a string that starts with three backticks followed by the string "python" and a newline character, then captures any sequence of characters (including whitespace and line breaks) until it encounters another newline and three backticks.
+    
+    return re.findall("`{3}python\n([\s\S]*?)\n`{3}", contents)
+
+def run(snippet: str, filename: str) -> SnippetExec:
+    '''
+    Save the code snippets into a file and run through it.
+    Return
+    - `SnippetExec` which tells us:
+        - Whether we are able to successfully run through the entire snippet
+        - Any errors
+        - Any output
+    '''
+    success = True
+    error = ""
+    output = ""
+
+    try:
+        # Create a temporary file with every code block from the Markdown file.
+        with open(filename, "w") as f:
+            f.write(snippet)
+
+        # Execute the code in the file and capture the output.
+        result = subprocess.run(["python3", filename], stdout=subprocess.PIPE, stderr=subprocess.PIPE, check=True, text=True)
+
+        # Store the output from execution.
+        output = result.stdout
+    except subprocess.CalledProcessError as e:
+        success = False
+        output = e.stdout
+        error = e.stderr
+    # Delete the temporary file
+    os.remove(filename)
+    return SnippetExec(snippet, success, output, error)
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser()
+
+    parser.add_argument(
+        "--docs-folder",
+        dest="docs_folder",
+        default="gitbook",
+        action="store",
+        help="Path to gitbook directory.",
+    )
+
+    parser.add_argument(
+        "--run-subset",
+        dest="run_subset",
+        default=False,
+        action="store_true",
+        help="Run a subset of the gitbook sections.",
+    )
+
+    parser.add_argument(
+        "--sections",
+        dest="sections",
+        default=[],
+        nargs='+',
+        action="store",
+        help="Subset of gitbook sections to run.",
```

Comment:
Can you note that this is the result of os.walk() and expects a list of tuples? Not super clear what the second value in each tuple is.

Reply:
This is actually expecting a list of strings. I convert it into a list of tuples with a dummy value to match the os.walk format (since os.walk is a generator and I only use the first item).